north-star-vision.md

```diff
@@ -0,0 +1,137 @@
+# North Star Vision
+
+## SIG Release Roadmap for 2021 and beyond
+
+This document contains the SIG Release Roadmap for 2021. The status tracking can
+be found at the bottom. More detailed information can be found on the [SIG
+Release][0] and [Release Engineering][1] project boards.
+
+[0]: https://github.com/orgs/kubernetes/projects/23
+[1]: https://github.com/orgs/kubernetes/projects/30
+
+### Primary Focus
+
+Establish a **consumable**, **introspectable**, and **secure** supply chain for
+Kubernetes. As a supply chain we understand the defining, building and
+publishing of Kubernetes related artifacts.
+
+1. **Consumable**: Improving the usability of artifacts by making their
+   consumption easier. This includes being process independent from external
+   companies like Google.
+1. **Introspectable**: It is clear for users at which point and how Kubernetes
+   artifacts are being built. This includes the documentation of all
+   deliverables as well as clarifying what we do not support.
+1. **Secure**: The artifacts we produce are verified for their integrity. This
+   applies to their functionality (we know what we deliver) as well as their
+   software security (we know when CVEs occur).
+
+### Deliverables
+
+The following deliverables are necessary to achieve the overall goal. Within
+the following listing, all deliverables are sorted by their priority.
+
+1. **Formalize supported release platforms (Introspectable)**
+
+   https://github.com/kubernetes/sig-release/issues/1337
+
+   Outcome: Definition of the life cycle for currently supported Kubernetes
+   artifacts and a guideline for the community about how to add new platforms.
+
+1. **Implement a Bill of Materials (BOM) for release artifacts (Introspectable /
+   Secure)**
+
+   https://github.com/kubernetes/release/issues/1837
+
+   Outcome: An automated formal verification of produced release artifacts for
+   every future release.
+
+1. **Enhance Kubernetes binary artifact management (Consumable)**
+
+   Outcome: Being able to promote files as artifacts and using this mechanism
+   for Kubernetes releases.
+
+1. **Define and collect metrics about Kubernetes releases (Introspectable)**
+
+   https://github.com/kubernetes/sig-release/issues/1527
+
+   Outcome: Being able to measure and interpret a set of defined metrics about
+   Kubernetes releases to associate actions with those.
+
+1. **Define and implement the release cadence survey (Introspectable)**
+
+   https://github.com/kubernetes/sig-release/issues/1526
+
+   Outcome: A regular survey evaluating the user experience of the current
+   release cadence.
+
+1. **Simplify CVE process for release management (Secure)**
+
+   https://github.com/kubernetes/sig-release/issues/896
+
+   https://github.com/kubernetes/release/issues/1354
+
+   Outcome: A documented and simple process for handling CVE information within
+   Kubernetes releases.
+
+1. **Establish Cluster API as first-class signal for upstream releases
```

Comment:
I checked this job https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-gcp#capg-conformance-v1alpha4-k8s-master&width=5 and there are several things to take into consideration:
1. it runs a script https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/master/scripts/ci-conformance.sh
that install KIND
2. it calls another script that creates some gcp infra with bash command "gcloud ..." and install packer https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/master/scripts/ci-e2e.sh
3. packer runs some ansible based on https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi
4. the it runs kind and does its stuff

It installs calico from master? 
> 
> # Deploy calico
> kubectl --kubeconfig="/tmp/kubeconfig" apply -f https://docs.projectcalico.org/manifests/calico.yaml
> configmap/calico-config created
> customresourcedefinition.apiextensions.k8s.io/bgpconfigurations.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/bgppeers.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/blockaffinities.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/clusterinformations.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/felixconfigurations.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/globalnetworkpolicies.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/globalnetworksets.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/hostendpoints.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/ipamblocks.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/ipamconfigs.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/ipamhandles.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/ippools.crd.projectcalico.org created
> customresourcedefinition.apiextensions.k8s.io/kubecontrollersconfigurations.crd.projectcalico.org created

I see many moving parts, from my experience it will take a lot to stabilise this job ... I only ask to not replace current jobs until this has demonstrated stability for at least one release cycle

Reply:
No objections to improving support for this mechanism. I agree with @aojea that the current mechanisms should remain until the new approach demonstrates stability for at least a release cycle. As an aside, I would have guessed this would have fallen under "Consumable", not sure how it relates to "Secure".

Reply:
as discussed recently with SIG Scale, the GCP provider for CAPI (CAPG) is not in a good state due to no active maintainers, that can dedicate a certain % of workday to the project.
the existing CI scripts and setup for CAPG are a best effort currently.